x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -400,21 +401,37 @@ export const endpointMiddlewareFactory: ImmutableMiddlewareFactory<EndpointState
     }
 
     // page activity log API
-    if (action.type === 'appRequestedEndpointActivityLog' && hasSelectedEndpoint(getState())) {
-      dispatch({
-        type: 'endpointDetailsActivityLogChanged',
-        // ts error to be fixed when AsyncResourceState is refactored (#830)
-        // @ts-expect-error
-        payload: createLoadingResourceState<ActivityLog>(getActivityLogData(getState())),
-      });
-
+    if (
+      action.type === 'endpointDetailsActivityLogUpdatePaging' &&
+      hasSelectedEndpoint(getState())
+    ) {
       try {
-        const { page, pageSize } = getActivityLogDataPaging(getState());
+        const { disabled, page, pageSize, startDate, endDate } = getActivityLogDataPaging(
+          getState()
+        );
+        const isInvalidDateRange =
+          startDate && endDate ? moment(startDate) > moment(endDate) : false;
```

Comment:
tip: `moment(startDate).isAfter(endDate)`

Reply:
and instead of checking `startDate && endDate` you could do `moment(startDate).isValid() && moment(endDate).isValid()`
What makes me think this could be written like:

```
const momentStartDate = moment(startDate);
const momentEndDate = moment(endDate);

const isValidDateRange = momentStartDate.isValid() && momentEndDate.isValid() && momentEndDate.isAfter(momentStartDate);

if (disabled || !isValidDateRange) {
...
```

I always try to avoid negative variables, for example, `isValidDateRange` instead of `isInvalidDateRange`

Reply:
Ooh nice with the helper methods from `moment`. I'll use those. 👍 

I tend to follow the same sort of rule in naming variables. 😄  I'll have to make an exception here though since I went with a name that's consistent with what the date picker prop name expects. It's much simpler to reason with when I read `isInvalid={isInvalidDateRange}`.

Reply:
done fe043611f64d025e4decb55acae48998328c8c6d. Also, note that I didn't use the `isValid()` method cause I think it's more performant just to use the native check if the values exist before passing them to `moment()`. https://github.com/elastic/kibana/pull/104085/commits/fe043611f64d025e4decb55acae48998328c8c6d#diff-986ae8726b37f91d8fdc9e624ae74e8705fdd4a2081aa855cde911311186418aR35